Jenkinsfile

```diff
@@ -0,0 +1,143 @@
+            withMavenEnv(["JAVA_OPTS=-Xmx1536m -Xms512m -XX:MaxPermSize=1024m",
+                          "MAVEN_OPTS=-Xmx1536m -Xms512m -XX:MaxPermSize=1024m"]) {
+                // Actually run Maven!
+                // The -Dmaven.repo.local=${pwd()}/.repository means that Maven will create a
+                // .repository directory at the root of the build (which it gets from the
+                // pwd() Workflow call) and use that for the local Maven repository.
+                sh "mvn -Pdebug -U clean install ${runTests ? '-Dmaven.test.failure.ignore=true -Dconcurrency=1' : '-DskipTests'} -V -B -Dmaven.repo.local=${pwd()}/.repository"
+            }
+        }
+
+        // Once we've built, archive the artifacts and the test results.
+        stage "Archive artifacts and test results"
+
+        archive includes: "**/target/*.jar, **/target/*.war, **/target/*.hpi"
+        if (runTests) {
+            step([$class: 'JUnitResultArchiver', healthScaleFactor: 20.0, testResults: '**/target/surefire-reports/*.xml'])
+        }
+
+        // And stash the jenkins.war for the next step
+        stash name: "jenkins.war", includes: "war/target/jenkins.war"
+    }
+}
+
+// Run the packaging build on a node with the "pkg" label.
+node('docker') {
+    // Add timestamps to logging output.
+    wrap([$class: 'TimestamperBuildWrapper']) {
+
+        // First stage here is getting prepped for packaging.
+        stage "packaging - docker prep"
+
+        // Docker environment to build packagings
+        dir('packaging-docker') {
+            git branch: packagingBranch, url: 'https://github.com/jenkinsci/packaging.git'
+            sh 'docker build -t jenkins-packaging-builder:0.1 docker'
+        }
+
+        stage "packaging - actually packaging"
+        // Working packaging code, separate branch with fixes
+        dir('packaging') {
+            git branch: packagingBranch, url: 'https://github.com/jenkinsci/packaging.git'
+            // Grab the war file from the stash - it goes to war/target/jenkins.war
+            unstash "jenkins.war"
+            sh "cp war/target/jenkins.war ."
+
+            sh 'docker run --rm -v "`pwd`":/tmp/packaging -w /tmp/packaging jenkins-packaging-builder:0.1 make clean deb rpm suse BRAND=./branding/jenkins.mk BUILDENV=./env/test.mk CREDENTIAL=./credentials/test.mk WAR=jenkins.war'
+
+            archive includes: "target/**/*"
+        }
+
+    }
+}
+
+stage "Package testing"
+
+if (runTests) {
+    // NOTE: As of now, a lot of package tests will fail. See https://issues.jenkins-ci.org/issues/?filter=15257 for
+    // possible open JIRAs.
+
+    // Basic parameters
+    String artifactName = (binding.hasVariable('artifactName')) ? artifactName : 'jenkins'
+    String jenkinsPort = (binding.hasVariable('jenkinsPort')) ? jenkinsPort : '8080'
+
+    // Set up
+    String debfile = "artifact://${env.JOB_NAME}/${env.BUILD_NUMBER}#target/debian/jenkins_2.0_all.deb"
```

Comment:
Ah, I think I'll try switching to using https://github.com/jenkinsci/pipeline-utility-steps-plugin/blob/master/docs/STEPS.md and the findFiles step there.